### PR TITLE
AnyCable Rails 1.3.0 and  Redis 5.0 compatibility

### PR DESCRIPTION
**Pull request by palkan**
- [x] Update Redis `pipeline` / `multi` usage (older API is deprecated)
- [x] Updated `istate` usage to be compatible with `anycable-rails` 1.3.0
  - [x]  Refactored tests to support legacy and newer anycable-rails
  - [x] Put both spec schemas into a single file (and add a namespace—required for the recent graphql ruby)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,19 +20,23 @@ jobs:
...
           - ruby: 2.7
             graphql: '~> 1.12.0'
             client_id: 'false'
             interpreter: yes
+            anycable_rails: '~> 1.2.0'
           - ruby: 2.6
             graphql: '~> 1.11.0'
             client_id: 'true'
             interpreter: no
+            anycable_rails: '~> 1.0'
```
Comment (Envek) [inline .github/workflows/test.yml:33]: Shouldn't it be `~> 1.0.0` to test older anycable versions?

**Review by Envek — COMMENTED** [review]
I can't find in what version of Redis.rb these new preferred API was introduced.

Maybe we need to restrict current version range (`>= 4.2.0`) even further in gemspec. Have you more info on that?

File: .github/workflows/test.yml (hunk 1 above)
Comment (palkan) [reply .github/workflows/test.yml:33]: The older version would be picked up automatically due according to the Ruby version

**Comment by palkan** [comment]
> I can't find in what version of Redis.rb these new preferred API was introduced. Have you more info on that?

I install fresh deps and found a lot of deprecations saying `Redis 5.0.0 is dropping bla-bla`.

See https://github.com/redis/redis-rb/blob/cc21d6762a63737969aa01139bb071da1f710b74/lib/redis/pipeline.rb#L55

**Review by Envek — APPROVED** [review]
Redis gem maintainer [says](https://github.com/redis/redis-rb/pull/1059#issuecomment-1064059208) that these APIs are available since 3.0.0 and we already have constraint for `>= 4.2.0`, so we should be fine.

**Pull request merged**